### ISSUE TITLE
Cover Block: Add Hover Reveal style

### ISSUE THREAD

**Issue by alaczek** [WordPress/gutenberg#20402]
**Is your feature request related to a problem? Please describe.**
Hovering over an image to reveal text/link is a popular web design pattern. It would be great if the Cover block enabled that by adding dedicated block style.

![epo251NU6k](https://user-images.githubusercontent.com/4550351/75122875-44891700-56f6-11ea-9193-f4e5b03aef04.gif)

The screenshot comes from [Dalston](https://wordpress.com/theme/dalston) theme ( [source](https://github.com/Automattic/themes/tree/master/dalston) ) where "Hover reveal" is added as a custom style to the Cover block.

Here's how it looks like in the editor:

![image](https://user-images.githubusercontent.com/4550351/75122894-96ca3800-56f6-11ea-8b77-993485539175.png)

The behaviour is the same in the editor and on the frontend  -- with "Hover reveal" style applied, the text shows up when hovering over an image. Clicking on the text makes it editable. Not sure if that particular behaviour in the editor would cause accessibility issues or not, but I'm sure we can iron out that part, if we agree that having this style added would be useful.

![Si2wN8bajX](https://user-images.githubusercontent.com/4550351/75123072-3f2ccc00-56f8-11ea-9f4e-5bd7f564993f.gif)

**Comment by jasmussen** [comment]
Nice ticket! This seems like a fine enhancement to add to the Cover block in my opinion. I'm assuming that focus also reveals?

**Comment by alaczek** [comment]
> I'm assuming that focus also reveals?
Do you mean when focusing with a keyboard? It does not in Dalston, although it definitely should.

**Comment by truchot** [comment]
I think it's possible to do this animation with the block cover and a bit of CSS, no?

**Comment by jasmussen** [comment]
> Do you mean when focusing with a keyboard? It does not in Dalston, although it definitely should.

Yep, I was thinking that if you tab into the block, the content should be revealed. Or even if you click on the hoverable area, that sets a lingering focus there, this should keep the content visible even as you mouse out and only disappear when you click somewhere else. 

To phrase it even simpler, instead of `:hover { ... }`, I'm suggesting `:hover, :focus { ... }` ;) — though we might have to throw in some `:focus-within` 😱 

> I think it's possible to do this animation with the block cover and a bit of CSS, no?

Absolutely. I read this ticket as posing the question: does this have so much value that it should be a toggle in the block, as opposed to require custom CSS?

**Comment by karmatosed** [comment]
I love this idea as feels like a great addition to the cover image block. Let's get this a 'needs dev' label and move into that.

**Comment by krzakx** [comment]
Hey, do we have any update on that?

**Comment by Draino** [comment]
Has this been implemented yet?

**Comment by vk17-starlord** [comment]
@karmatosed , @jordesign , @jasmussen 

Hi everyone 👋
Just checking in — is this issue still open and relevant? If so, I’d be happy to work on a PR to add the “Hover reveal” style enhancement to the Cover block. Let me know if this is still something you’d like to see implemented!

Thanks!
